### PR TITLE
Fix SAML Single Logout

### PR DESCRIPTION
**Pull request by coudot**
Hello,

we are using NextCloud as SAML Service Provider with [LemonLDAP::NG](https://www.lemonldap-ng.org) as SAML Identity Provider. LemonLDAP::NG is using [Lasso](https://lasso.entrouvert.org/) SAML library, which is conform to SAML 2.0 specifications.

We have an issue with Single Logout (SAML SLO) initiated by NextCloud. The NameID set in LogoutRequest is not valid, as NameID Format, NameID NameQualifer and NameID SP NameQualifier are not set if they exist if the NameID transmitted by SAML IDP in AuthnResponse.

In the [SAML core specification](https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf), we can read:
> <saml:BaseID> or <saml:NameID> or <saml:EncryptedID> [Required]The identifier and associated attributes (in plaintext or encrypted form) that specify the principal ascurrently recognized by the identity and service providers prior to this request.  (For more informationon this element, see Section 2.2.)

It means the NameID markup must be the same as the one given by the IDP at authentication step.

You will find our patch to fix this issue, which was tested on our side.

Please let us know if you need something else to include this fix in your code base.

### CODE REVIEW

**Comment by coudot** [comment]
Hello,

is there someone in the development team who could review this patch? I am available for any question.

Clément.

File: 3rdparty/vendor/onelogin/php-saml/README.md [hunk 1]
```diff
@@ -1169,7 +1169,7 @@ Main class of OneLogin PHP Toolkit
  * `getNameId` - Returns the nameID
  * `getNameIdFormat` - Gets the NameID Format provided by the SAML response from the IdP.
  * `getNameIdNameQualifier` - Gets the NameID NameQualifier provided from the SAML Response String.
- * `getNameIdNameSPQualifier` - Gets the NameID SP NameQualifier provided from the SAML Response String.
```
Comment (juliushaertl) [inline 3rdparty/vendor/onelogin/php-saml/README.md]: Could you remove the changes from 3rdparty/ ? Documentation is already fixed upstream with https://github.com/onelogin/php-saml/commit/ead86cf9ac7d2218a470aff029782db3eac13032 and we get that once we update the vendor package again.

**Comment by coudot** [comment]
Hello @juliushaertl, thanks a lot for your feedback!

I applied your suggestions (rebase and revert the doc modification), hope all is fine.

**Comment by coudot** [comment]
Thanks a lot for the merge!

**Pull request merged**